### ISSUE TITLE
DRONE_WORKSPACE and CI_WORKSPACE env variable behaves weird

### ISSUE THREAD

**Issue by choffmeister** [drone/drone#2466]
Hi. I am using the latest drone docker image `0.8.6` and experience some weird behaviour with the `DRONE_WORKSPACE`/`CI_WORKSPACE` variables. They seem to be only injected as environment variables when running the image, but are not in the "metadata" (so cannot be used for string operations as explained [here](http://readme.drone.io/0.5/usage/environment-reference/)).

Here a quick test case to reproduce:

```yaml
pipeline:
  test1:
    image: busybox:latest
    environment:
      - DRONE_WORKSPACE2=${DRONE_WORKSPACE}
    commands:
      # works
      - echo $DRONE_WORKSPACE
      # is empty, but I would expected it to be filled, too
      - echo $DRONE_WORKSPACE2
      # is empty, but I would expected it to be filled, too
      - echo ${DRONE_WORKSPACE}

  test2:
    image: busybox:latest
    environment:
      - CI_WORKSPACE2=${CI_WORKSPACE}
    commands:
      # works
      - echo $CI_WORKSPACE
      # is empty, but I would expected it to be filled, too
      - echo $CI_WORKSPACE2
      # is empty, but I would expected it to be filled, too
      - echo ${CI_WORKSPACE}
```

Is this intentional?

**Comment by bradrydzewski** [comment]
see the section on interpolation and escaping http://docs.drone.io/environment/

**Comment by choffmeister** [comment]
@bradrydzewski I read that. Let me rephrase my point:

* Why is `DRONE_WORKSPACE` an environment variable and so cannot be used for interpolation,
* but `DRONE_COMMIT_SHA` is some thing different and so can be used for interpolation.

The `DRONE_WORKSPACE` value is also something fixed (at least it can be inferred from configration and does not change while running the build) and so I don't understand, why those two should behave differently.

Example:

```yaml
pipeline:
  test1:
    image: busybox:latest
    environment:
      - DRONE_WORKSPACE2=${DRONE_WORKSPACE}
      - DRONE_COMMIT_SHA2=${DRONE_COMMIT_SHA}
    commands:
      # works
      - echo $DRONE_WORKSPACE
      # is empty, but I would expected it to be filled, too
      - echo $DRONE_WORKSPACE2
      # is empty, but I would expected it to be filled, too
      - echo ${DRONE_WORKSPACE}
      # works
      - echo $DRONE_COMMIT_SHA
      # works
      - echo $DRONE_COMMIT_SHA2
      # works
      - echo ${DRONE_COMMIT_SHA}
```

```
+ echo $DRONE_WORKSPACE
/drone/src/git.codingwerft.de/airfocus/airfocus-server
+ echo $DRONE_WORKSPACE2

+ echo

+ echo $DRONE_COMMIT_SHA
573e8a09461244a8f2e3c411dc79db4af76db41f
+ echo $DRONE_COMMIT_SHA2
573e8a09461244a8f2e3c411dc79db4af76db41f
+ echo 573e8a09461244a8f2e3c411dc79db4af76db41f
573e8a09461244a8f2e3c411dc79db4af76db41f
```

**Comment by bradrydzewski** [comment]
DRONE_WORKSPACE is not available for interpolation. See https://github.com/drone/drone/issues/1322

**Comment by choffmeister** [comment]
@bradrydzewski Got it. Thanks
